### ISSUE TITLE
yii2 tests postgres tests fail for bigint and smallint types

### ISSUE THREAD

**Issue by gvasilopulos** [yiisoft/yii2#13511]
### What steps will reproduce the problem?
in @vendor/yiisoft/yii2-dev run phpunit --log-junit pgtests.xml  --group=pgsql
### What is the expected result?
pass of tests
### What do you get instead?
see test results in xml and raw output format here 
https://gist.github.com/gvasilopulos/502288deda6f9e364abb03e5394cdd82

### Additional info
using php 5.6.25 with 
"PostgreSQL 9.5.3 on i486-slackware-linux-gnu, compiled by gcc (GCC) 5.3.0, 32-bit"
on Slackware linux 14.2  32 bit 
Linux daidalos 4.4.38-smp

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.11
| PHP version      | 5.6.25
| Operating system | Slackware linux 14.2  32 bit Linux daidalos 4.4.38-smp

**Comment by samdark** [comment]
Reproduced with `./vendor/bin/phpunit tests/framework/db/pgsql/SchemaTest.php`. 2.0.10 and 2.0.9 are behaving the same way.

**Comment by cebe** [comment]
what is different from travis here? postgres version?=

**Comment by gvasilopulos** [comment]
pg version is 9.5.3 it is a 32bit machine 
some more info which could be significant 
the number 9.2233720368548E+18 returns  
ERROR:  bigint out of range

********** Error **********

ERROR: bigint out of range
SQL state: 22003 
(32bit limit?) on a query with pgadmin 
while the number 
9.2233720368548E+17 gets inserted without a problem in a query with pgadmin, strange thing is that both numbers appear in single quotes in the produced query of the test and that is exactly the reason not to pass the test , see the test query: 
INSERT INTO "type" ("bigint_col", "int_col", "char_col", "float_col", "bool_col") VALUES ('9.2233720368548E+17', 1, 'a', 0.1, TRUE) <--not working with syntax error
while it should be 
INSERT INTO "type" ("bigint_col", "int_col", "char_col", "float_col", "bool_col") VALUES (9.2233720368548E+17, 1, 'a', 0.1, TRUE) <-- tested, works in pg,
is it possible when number are large enough to be expressed as an E+XX value something in pdo or yii supposes string value ?

**Comment by samdark** [comment]
Yes, that's 32 bit limitation.

**Comment by gvasilopulos** [comment]
yes but this isn't 9.2233720368548E+17 . The error with 9.2233720368548E+18 is normal and in my case unavoidable but for 9.2233720368548E+17  shouldn't happen..

**Comment by terabytesoftw** [comment]
Thanks for the report.

I re-checked this on current Yii2 master (March 2026) and could not reproduce it on the supported baseline (PHP >= `7.4`). The related PostgreSQL bigint test passes on current Yii2.

This issue was originally reported for Yii `2.0.11` / PHP `5.6` on `32-bit x86`. The failing case is tied to `32-bit` numeric limitations (large bigint literals becoming floating-point scientific notation), which is environment-specific rather than an actionable bug in
current Yii2.
